File: VB/WindowsApplication53/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms

Namespace WindowsApplication53

    Public Partial Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs)
            PopulateTable()
            pivotGridControl1.RefreshData()
            pivotGridControl1.BestFit()
            pivotGridControl1.FieldsCustomization(panelControl1)
        End Sub

        Private Sub PopulateTable()
            Dim myTable As DataTable = dataSet1.Tables("Data")
            myTable.Rows.Add(New Object() {"Aaa", Date.Today, Nothing})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddDays(1), Nothing})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today, 0})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddDays(1), 12})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today.AddDays(1), 0})
            myTable.Rows.Add(New Object() {"Ddd", Date.Today.AddYears(1), Nothing})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddYears(1), 4})
            myTable.Rows.Add(New Object() {"Aaa", Date.Today.AddYears(1).AddDays(1), 2})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddYears(1), 0})
            myTable.Rows.Add(New Object() {"Bbb", Date.Today.AddDays(1).AddYears(1), 0})
            myTable.Rows.Add(New Object() {"Ccc", Date.Today.AddYears(1), Nothing})
        End Sub

        Private Sub pivotGridControl1_CustomCellDisplayText(ByVal sender As Object, ByVal e As DevExpress.XtraPivotGrid.PivotCellDisplayTextEventArgs)
            If Convert.ToDecimal(e.Value) = 0 AndAlso IsNullValue(e.CreateDrillDownDataSource(), e.DataField) Then e.DisplayText = "null"
        End Sub

        Private Function IsNullValue(ByVal ds As DevExpress.XtraPivotGrid.PivotDrillDownDataSource, ByVal field As DevExpress.XtraPivotGrid.PivotGridField) As Boolean
            If ds.RowCount = 0 Then Return False
            For i As Integer = 0 To ds.RowCount - 1
                If Not Equals(ds(i)(field), Nothing) Then Return False
            Next

            Return True
        End Function
    End Class
End Namespace

File: CS/WindowsApplication53/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication53
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }
        private void Form1_Load(object sender, EventArgs e)
        {
            PopulateTable();
            pivotGridControl1.RefreshData();
            pivotGridControl1.BestFit();
            pivotGridControl1.FieldsCustomization(panelControl1);
            
        }
        private void PopulateTable()
        {
            DataTable myTable = dataSet1.Tables["Data"];
            myTable.Rows.Add(new object[] {"Aaa", DateTime.Today, null});
            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddDays(1), null });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today, 0 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddDays(1), 12 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today.AddDays(1), 0 });

            myTable.Rows.Add(new object[] { "Ddd", DateTime.Today.AddYears(1), null });

            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddYears(1), 4 });
            myTable.Rows.Add(new object[] { "Aaa", DateTime.Today.AddYears(1).AddDays(1), 2 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddYears(1), 0 });
            myTable.Rows.Add(new object[] { "Bbb", DateTime.Today.AddDays(1).AddYears(1), 0 });
            myTable.Rows.Add(new object[] { "Ccc", DateTime.Today.AddYears(1), null });
        }

        private void pivotGridControl1_CustomCellDisplayText(object sender, DevExpress.XtraPivotGrid.PivotCellDisplayTextEventArgs e)
        {
            if (Convert.ToDecimal(e.Value) ==  0 && IsNullValue(e.CreateDrillDownDataSource(), e.DataField))
                e.DisplayText = "null";
        }

        private bool IsNullValue(DevExpress.XtraPivotGrid.PivotDrillDownDataSource ds, DevExpress.XtraPivotGrid.PivotGridField field)
        {
            if (ds.RowCount == 0) return false;
            for (int i = 0; i < ds.RowCount ; i++)
            {
                if (!Equals(ds[i][field], null)) return false;
            }
            return true;
        }


       
    }
}